tools/codegen/cmd/featuregate-test-analyzer.go

```diff
@@ -429,20 +446,23 @@ func testResultByName(results []TestResults, testName string) *TestResults {
 }
 
 func listTestResultFor(featureGate string, clusterProfiles sets.Set[string]) (map[JobVariant]*TestingResults, error) {
-	testPattern := fmt.Sprintf("[OCPFeatureGate:%s]", featureGate)
-	fmt.Printf("Query component readiness for all test run results for pattern %q on clusterProfile %q\n", testPattern, sets.List(clusterProfiles))
+	fmt.Printf("Query component readiness for all test run results for feature gate %q on clusterProfile %q\n", featureGate, sets.List(clusterProfiles))
 
 	results := map[JobVariant]*TestingResults{}
 
-	jobVariantsToCheck := []JobVariant{}
-	if clusterProfiles.Has("Hypershift") {
-		jobVariantsToCheck = append(jobVariantsToCheck, requiredHypershiftJobVariants...)
+	var jobVariantsToCheck []JobVariant
+	if clusterProfiles.Has("Hypershift") && !nonHypershiftPlatforms.MatchString(featureGate) {
+		jobVariantsToCheck = append(jobVariantsToCheck, filterVariants(featureGate, requiredHypershiftJobVariants, false)...)
 	}
 	if clusterProfiles.Has("SelfManagedHA") {
-		jobVariantsToCheck = append(jobVariantsToCheck, requiredSelfManagedJobVariants...)
-	}
-	if len(jobVariantsToCheck) == 0 {
-		jobVariantsToCheck = append(jobVariantsToCheck, requiredSelfManagedJobVariants...)
+		// See if the feature gate is specific to any optional platform
+		optionalPlatformVariants := filterVariants(featureGate, optionalSelfManagedPlatformVariants, true)
```

Comment:
oh it isn't. I've misunderstood.  This is a sometimes list versus the rest.  Thinking about the factoring here.